### ISSUE TITLE
OSD on PVC should have a default value "metadata" or "data"

### ISSUE THREAD

**Issue by rohantmp** [rook/rook#4924]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When creating a CephCluster that doesn't specify `storageClassDeviceSets.volumeClaimTemplate.metadata.name`, the PVC is created, but the OSD prepare/OSD pod never get created.

The following log message indicates why this occurs:
```
2020-02-27 14:01:14.732115 I | op-osd: successfully provisioned pvc "ocs-deviceset-2--0-lgr76" for VolumeClaimTemplates "" for storageClassDeviceSet "ocs-deviceset-2" of set 0
```
**Expected behavior:**
If the value is omitted, the OSD should be considered a "data" volume by default
**How to reproduce it (minimal and precise):**
- Deploy rook based on latest master
- Create a CephCluster that doesn't specify `storageClassDeviceSets.volumeClaimTemplate.metadata.name`

The problem doesn't appear if you specify `storageClassDeviceSets.volumeClaimTemplate.metadata.name`.

**Comment by rohantmp** [comment]
@travisn @leseb I can take a look at this in a couple of days if necessary.

**Comment by sp98** [comment]
@rohantmp  Using a default value looks like a good option. But could it just be a mandatory field that user has to provide every time?

**Comment by leseb** [comment]
The default template sets a "name", I'd think people would just rely on the template (it also has comments too).
I believe it's acceptable to default to "data" if nothing is set.

**Comment by travisn** [comment]
Sounds good to default to "data", otherwise we would probably have backward compatibility issues since it wasn't required in the previous release IIRC.

**Comment by travisn** [comment]
This was already fixed by #5524
